File: my-folder/0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if(s.length() == 0) {
            return 0;
        }
        int ans = 1;
        for(int i = 0; i < s.length();i++) {
            Set<Character> set = new HashSet<>();
            set.add(s.charAt(i));
            for (int j = i + 1; j < s.length();j++) {
                if(!set.contains(s.charAt(j))) {
                    if (j - i + 1 > ans) {
                        ans = j - i + 1;
                    }
                } else {
                    break;
                }
                set.add(s.charAt(j));
            }
        }
        return ans;
    }
}

File: my-folder/0003-longest-substring-without-repeating-characters/solution.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int left = 0;
        int right = 0;
        var usedChars = new HashSet<char>();
        int ans = 0;
        while (right < s.Length) {
            while(usedChars.Contains(s[right])) {
                usedChars.Remove(s[left]);
                left++;
            }
            ans = Math.Max(ans, right - left + 1);
            usedChars.Add(s[right]);
            right++;
        }
        return ans;
    }
}
